### PR TITLE
Add option to condense conftest output

### PR DESCRIPTION
**Pull request by joejstuart**
This adds an option to shorten the report output. The output is restricted to one error message per short_name that's defined in the policy annotation.

### CODE REVIEW

File: internal/applicationsnapshot/report.go [hunk 1]
```diff
@@ -30,13 +31,50 @@ type Component struct {
 	Success    bool            `json:"success"`
 }
 
-type report struct {
+type fullReport struct {
```
Comment (zregvart) [inline internal/applicationsnapshot/report.go:34]: I think the struct can still be called report...
```suggestion
type report struct {
```

File: internal/applicationsnapshot/report.go [hunk 2]
```diff
@@ -30,13 +31,50 @@ type Component struct {
...
 	Success    bool        `json:"success"`
 	Components []Component `json:"components"`
 }
 
+type shortReport struct {
+	Components []shortComponent `json:"components"`
+	Success    bool             `json:"success"`
+}
+
+type shortComponent struct {
```
Comment (zregvart) [inline internal/applicationsnapshot/report.go:44]: Maybe:
```suggestion
type componentSummary struct {
```

File: internal/applicationsnapshot/report.go [hunk 3]
```diff
@@ -30,13 +31,50 @@ type Component struct {
...
+	TotalViolations int                 `json:"total_violations"`
+	TotalWarnings   int                 `json:"total_warnings"`
+}
+
+func NewReport(components []Component, shortReport bool) (string, error, bool) {
+	var j []byte
+	var err error
+	var success bool
+	if shortReport {
+		report := condensedReport(components)
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply internal/applicationsnapshot/report.go]: absolutely, just pushed the changes. thanks!

File: cmd/validate_image.go [hunk 4]
```diff
@@ -98,14 +99,22 @@ Return a non-zero status code on validation failure:
...
+
 Use an EnterpriseContractPolicy resource from the currently active kubernetes context:
 
   ec validate image --image registry/name:tag --policy my-policy
 
 Use an EnterpriseContractPolicy resource from a different namespace:
 
   ec validate image --image registry/name:tag --policy my-namespace/my-policy
+  
+Use an EnterpriseContractPolicy resource from a different namespace:
```
Comment (lcarva) [inline cmd/validate_image.go:114]: This example looks identical to the one above (including the description)

File: cmd/validate_image.go [hunk 5]
```diff
@@ -220,6 +230,7 @@ Use an inline EnterpriseContractPolicy spec
 		"write output to a file. Use empty string for stdout, default behavior")
 	cmd.Flags().BoolVarP(&data.strict, "strict", "s", data.strict,
 		"return non-zero status on non-successful validation")
+	cmd.Flags().BoolVarP(&data.summary, "summary", "c", data.summary, "print condensed output of conftest")
```
Comment (lcarva) [inline cmd/validate_image.go:233]: I think mentioning `conftest` here is not very helpful to the user. Maybe, something like `condense violation and warning messages`?

Comment (joejstuart) [reply cmd/validate_image.go]: I agree. updated.

File: cmd/validate_image.go (hunk 4 above)
Comment (joejstuart) [reply cmd/validate_image.go]: yep, copy/paste mistake.

**Review by zregvart — APPROVED** [review]
Looks good! I think one semi-obvious followup we can do is to have a more elaborate `--output` that not only takes into account different formats (JSON/YAML) but also can do `summary`, `template`, `jsonpath` -- similar to kubectl (created [HACBS-1353](https://issues.redhat.com/browse/HACBS-1353))

**Pull request merged**